GeeksCoreLibrary/Modules/PostalServices/PostNL/Controllers/PostNLController.cs

```diff
@@ -0,0 +1,231 @@
+                        {
+                            new ShipmentModel
+                            {
+                                ProductCodeDelivery = settings.ProductCode,
+                                Addresses = new List<AddressModel> { shippingAddress },
+                                Barcode = barcode,
+                                Contacts = new List<ContactModel>
+                                {
+                                    new ContactModel
+                                    {
+                                        Email = orderDetails.GetDetailValue("email"),
+                                        SmsNumber = orderDetails.GetDetailValue("phone")
+                                    }
+                                },
+                                Remark = orderId
+                            }
+                        }
+                };
+                if (shippingLocation == PostNLService.ShippingLocations.Global)
+                {
+                    postNlRequest.Shipments.First().Customs = new CustomsModel
+                    {
+                        Content = new List<CustomsContentModel>(),
+                        Currency = "EUR",
+                        HandleAsNonDeliverable = "false",
+                        Invoice = "true",
+                        InvoiceNr = orderId,
+                        ShipmentType = "Commercial Goods"
+                    };
+                    var orderLines = await wiserItemsService.GetLinkedItemDetailsAsync(UInt64.Parse(orderId), 5002, "orderline");
+                    foreach (WiserItemModel orderLine in orderLines)
+                    {
+                        postNlRequest.Shipments.First()
+                            .Customs.Content.Add(new CustomsContentModel
+                            {
+                                Description = orderLine.GetDetailValue("title"),
+                                CountryOfOrigin = "NL",
+                                HSTariffNr = "621112",
+                                Quantity = orderLine.GetDetailValue("quantity"),
+                                Value = orderLine.GetDetailValue("price").Replace(",", "."),
+                                Weight = "500"
+                            });
+                    }
+                }
+
+                var postNlResponse = await postNlService.CreateTrackTraceLabel(orderId, postNlRequest);
+                if (postNlResponse?.ResponseShipments == null || !postNlResponse.ResponseShipments.Any())
+                {
+                    result.Add($"Order {orderId}: Er is iets fout gegaan met de koppeling met de PostNL API.");
+                    continue;
+                }
+
+                barcode = postNlResponse.ResponseShipments.First().Barcode;
+
+                orderDetails.SetDetail("postnl_barcode", barcode);
+                orderDetails.SetDetail("country_code", GetCountryCode(orderDetails, "country"));
```

Comment:
Dit kan nooit goed werken. De functie `GetCountryCode` is een async functie. Graag hier een `await` voor zetten